deps/rabbit/src/mc_amqp.erl

```diff
@@ -63,6 +63,10 @@
 init(Sections) when is_list(Sections) ->
     Msg = decode(Sections, #msg{}),
     init(Msg);
+init(Amqp10Msg) when element(1, Amqp10Msg) =:= amqp10_msg ->
+    [#'v1_0.transfer'{}| AmqpRecords] = amqp10_msg:to_amqp_records(Amqp10Msg),
```

Comment:
This conversions seems a little bit hacky, though we could not come up with a better way. 

By adding the `v1_0.transfer` record match at the beginning of the list, it effectively drops that list item.

Reply:
This introduce an implicit dependency on the AMQP 1.0 client to RabbitMQ core. We plan on making AMQP 1.0 support a core feature later this year (after https://github.com/rabbitmq/rabbitmq-server/pull/9022 is merged) but we cannot accept a dependency on an AMQP 1.0 client.

Just like at some point `rabbit_core` had to be introduced to avoid a dependency on `amqp_client` for AMQP 0-9-1.